src/aics-image-viewer/components/Toolbar/index.tsx

```diff
@@ -41,116 +42,214 @@ interface ToolbarProps {
   };
 }
 
-export default function Toolbar(props: ToolbarProps): React.ReactElement {
-  const { renderConfig, showAxes, showBoundingBox, autorotate } = props;
+interface ToolbarState {
+  scrollMode: boolean;
+  scrollBtnLeft: boolean;
+  scrollBtnRight: boolean;
+}
+
+export default class Toolbar extends React.Component<ToolbarProps, ToolbarState> {
+  barRef: React.RefObject<HTMLDivElement>;
+  leftRef: React.RefObject<HTMLDivElement>;
+  rightRef: React.RefObject<HTMLDivElement>;
+  centerRef: React.RefObject<HTMLDivElement>;
+
+  constructor(props: ToolbarProps) {
+    super(props);
+    this.state = {
+      scrollMode: false,
+      scrollBtnLeft: false,
+      scrollBtnRight: false,
+    };
+    this.barRef = React.createRef();
+    this.leftRef = React.createRef();
+    this.rightRef = React.createRef();
+    this.centerRef = React.createRef();
+
+    window.addEventListener("resize", this.checkSize);
+    this.checkSize();
+  }
 
-  const twoDMode = props.mode !== ViewMode.threeD;
+  checkSize = debounce((): void => {
+    const { leftRef, centerRef, rightRef, barRef } = this;
+    const leftRect = leftRef.current!.getBoundingClientRect();
+    const centerRect = centerRef.current!.getBoundingClientRect();
+    const rightRect = rightRef.current!.getBoundingClientRect();
 
-  const renderGroup1 =
-    renderConfig.viewModeRadioButtons || renderConfig.resetCameraButton || renderConfig.autoRotateButton;
-  const renderGroup4 = renderConfig.showAxesButton || renderConfig.showBoundingBoxButton;
+    if (this.state.scrollMode) {
+      // Leave scroll mode if there is enough space for centered controls not to overlap left/right-aligned ones
+      const barWidth = barRef.current!.getBoundingClientRect().width;
+      const requiredWidth = Math.max(leftRect.width, rightRect.width) * 2 + centerRect.width + 30;
```

Comment:
I suggest making the 30 a named symbolic constant.  It looks like a magic number.